client_server_test.go

```diff
@@ -659,3 +661,31 @@ func TestSocksProxyDial(t *testing.T) {
 	defer ws.Close()
 	sendRecv(t, ws)
 }
```

Comment:
Add test similar to TestHandshakeTimeout, but set deadline through context instead of Dialer.HandshakeTimeout.